internal_ws/yktrace/src/tir.rs

```diff
@@ -153,10 +154,23 @@ impl<'a, 'm> TirTrace<'a, 'm> {
                             idx: rnm.rename_irplace(idx, &body),
                             scale: *scale,
                         },
-                        Statement::Store(dst, src) => Statement::Store(
-                            rnm.rename_irplace(dst, &body),
-                            rnm.rename_irplace(src, &body),
-                        ),
+                        Statement::Store(dst, src) => {
+                            if matches!(dst, IRPlace::Val { local, .. } if local == &sir::RETURN_LOCAL)
```

Comment:
I think you can reuse the `in_interp_step` variable instead of querying the body again?

Does the return value have to be constant? If so, does the compiler check that?

Reply:
1) I think `in_interp_step` means we tracing, it doesn't mean we are processing statements of the `interp_step` function, i.e. we could also be in another function that is being inlined. So we need to check the body again, since we only want to turn `return false` into `LoopEnd` if we are looking at the return of `interp_step` function.

Reply:
> I think in_interp_step means we tracing, it doesn't mean we are processing statements of the interp_step function.

I just checked. I think you are right. It's true between the call and return (including callees) to `interp_step`.

Reply:
About 2:

```
if matches!(src, IRPlace::Const { val: Constant::Bool(false), ..})
```

What if the boolean being returned is not a constant? Do we allow this?

```
fn interp_step(...) -> bool {
    let ret = false;
    ...
    if (somethig-dynamic) {
        ret = true;
    }
    ret;
}
```

Reply:
Fair doos. I hadn't thought about that. The question is: can we get the value out of the local at TIR time?

Reply:
After thinking about this a bit more: We could do some very basic static analysis to see what value the returned local has, but it's not a solution for the general case. As a very stupid example: the user could read the `bool` from a file or calculate it randomly in which case we'll never know. So maybe the right thing to do here is to enforce that they must return a constant.

Reply:
Then we can resolve the comment below by causing a panic or similar which gives the user a hint that the return value must be a constant.

Reply:
I agree that we should enforce a constant return value.

Reply:
Fixed in d7f02a4.

Reply:
The only problem with this fix is that it's only checked at JIT compile time. But I'm not sure how easy it will be to check this at rustc compile time.

Reply:
You'd have to walk the MIR looking for returns. I think it's OK to check at runtime for now. Perhaps raise an issue?